specs/phase1/shard-transition.md

```diff
@@ -44,17 +44,33 @@ def compute_shard_transition_digest(beacon_state: BeaconState,
 
 ```python
 def verify_shard_block_message(beacon_state: BeaconState,
-                               shard_state: ShardState,
+                               shard_parent_state: ShardState,
                                block: ShardBlock,
                                slot: Slot,
-                               shard: Shard) -> bool:
-    assert block.shard_parent_root == shard_state.latest_block_root
-    assert block.slot == slot
-    next_slot = Slot(beacon_state.slot + 1)
+                               shard: Shard,
+                               beacon_parent_slot: Slot=None) -> bool:
+    # Check `shard_parent_root` field
+    assert block.shard_parent_root == shard_parent_state.latest_block_root
+    # Check `beacon_parent_root` field
+    if beacon_parent_slot is None:
```

Comment:
I think we can remove `beacon_parent_slot` entirely and remove both of the outer block `if` statements to just have the following:

```python
    beacon_parent_block_header = beacon_state.latest_block_header.copy()
    if beacon_parent_block_header.state_root == Root():
        beacon_parent_block_header.state_root = hash_tree_root(beacon_state)
        beacon_parent_root = hash_tree_root(beacon_parent_block_header)
    else:
        beacon_parent_root = get_block_root_at_slot(beacon_state, beacon_parent_slot)
    assert block.beacon_parent_root == beacon_parent_root
```

That way, we are just making sure that the `block.beacon_parent_root` is coherent wrt the passed in `beacon_state`.